#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_fft_with_empty_matrix.dpatch by  <root@asterix.num.uni-sb.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Bug fix for #423278; synced with upstream CVS 

@DPATCH@
diff -urNad octave2.9-2.9.10~/src/DLD-FUNCTIONS/fft.cc octave2.9-2.9.10/src/DLD-FUNCTIONS/fft.cc
--- octave2.9-2.9.10~/src/DLD-FUNCTIONS/fft.cc	2006-05-19 05:32:18.000000000 +0000
+++ octave2.9-2.9.10/src/DLD-FUNCTIONS/fft.cc	2007-05-16 13:47:31.000000000 +0000
@@ -99,7 +99,7 @@
   if (dim < 0)
     {
       for (octave_idx_type i = 0; i < dims.length (); i++)
-	if ( dims(i) > 1)
+	if (dims(i) > 1)
 	  {
 	    dim = i;
 	    break;
@@ -115,8 +115,8 @@
   else
     dims (dim) = n_points;
 
-  if (dims.all_zero () || n_points == 0)
-    return octave_value (Matrix ());
+  if (dims.any_zero () || n_points == 0)
+    return octave_value (NDArray (dims));
 
   if (arg.is_real_type ())
     {
@@ -146,6 +146,20 @@
   return retval;
 }
 
+/*
+
+%!error(fft())
+%!assert(fft([]), [])
+%!assert(fft(zeros(10,0)), zeros(10,0))
+%!assert(fft(zeros(0,10)), zeros(0,10))
+%!assert(fft(0), 0)
+%!assert(fft(1), 1)
+%!assert(fft(1), 1)
+%!assert(fft(ones(2,2)), [2,2; 0,0])
+%!assert(fft(eye(2,2)), [1,1; 1,-1])
+
+*/
+
 
 DEFUN_DLD (fft, args, ,
   "-*- texinfo -*-\n\
